### PR TITLE
BaSP - MR-100: Members redux implementation

### PR DESCRIPTION
**Pull request by JuanIgnacioSobral**
Fully functional redux implementation

### CODE REVIEW

File: src/Components/Members/Edit/EditMembers.jsx [hunk 1]
```diff
@@ -44,20 +49,31 @@ const MembersEdit = () => {
...
-    try {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/members/${id}`);
-      const data = await response.json();
-      setMember(data.data);
-    } catch (error) {
-      console.error(error);
-    }
-  };
+  useEffect(() => {
+    console.log(data.getById.data);
```
Comment (MatiasSauvanet) [inline src/Components/Members/Edit/EditMembers.jsx:53]: Delete this console log

File: src/redux/members/thunks.js [hunk 2]
```diff
@@ -1,12 +1,135 @@
...
+export const getMemberById = (id) => {
+  return async (dispatch) => {
+    try {
+      dispatch(getMemberByIdPending());
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/members/${id}`);
+      const data = await response.json();
+      if (data.error) {
+        throw new Error(data.message);
+      }
+      console.log(data.data);
```
Comment (MatiasSauvanet) [inline src/redux/members/thunks.js:45]: Same here

File: src/redux/members/thunks.js [hunk 3]
```diff
@@ -1,12 +1,135 @@
...
+      dispatch(getMemberByIdSuccess(data.data));
+      console.log(data.data);
```
Comment (MatiasSauvanet) [inline src/redux/members/thunks.js:47]: Same here

**Review by MatiasSauvanet — COMMENTED** [review]
-When you want to add, there is not any error modals about incomplete inputs, you should add them. 
-In the edit, this works fine, but the errors modals should be red. Besides, when you successfully edit a member, the success modal will not go automaticly and dont redirect the page.

**Review by keiler-art — COMMENTED** [review]
The form is stuck to the nav bar, it should include a margin similar to the other entities.

**Review by Milicerro — COMMENTED** [review]
-CSS: add 0.8 oppacity to the blue background color.

File: src/Components/Members/Edit/EditMembers.jsx (hunk 1 above)
Comment (JuanIgnacioSobral) [reply src/Components/Members/Edit/EditMembers.jsx:53]: Removed!

File: src/redux/members/thunks.js (hunk 2 above)
Comment (JuanIgnacioSobral) [reply src/redux/members/thunks.js:45]: Done

File: src/redux/members/thunks.js (hunk 3 above)
Comment (JuanIgnacioSobral) [reply src/redux/members/thunks.js:47]: Also removed

**Comment by SantiH97** [comment]
yo should add this to your .container class
background-size: cover;
background-repeat: no-repeat;

**Comment by SantiH97** [comment]
The delete modal's title should be "Delete member" and its text be "are you sure you want to delete this member" you currently only have a tittle

File: src/redux/members/reducer.js [hunk 4]
```diff
@@ -19,4 +29,129 @@ const membersReducer = (state = INITIAL_STATE, action) => {
   }
 };
 
+const getByIdReducer = (state = initialState, action) => {
```
Comment (DelfiLT) [inline src/redux/members/reducer.js:32]: You have to do it all in one reducer, It canta be separated

File: src/redux/members/reducer.js [hunk 5]
```diff
@@ -19,4 +29,129 @@ const membersReducer = (state = INITIAL_STATE, action) => {
...
+        message: action.payload,
+        isPending: false,
+        error: true
+      };
+    default:
+      return state;
+  }
+};
+
+const membersReducer = combineReducers({
```
Comment (DelfiLT) [inline src/redux/members/reducer.js:149]: You should only export one reducer

File: src/redux/members/thunks.js [hunk 6]
```diff
@@ -1,12 +1,135 @@
...
+        throw new Error(data.message);
+      }
+      dispatch(getMembersSuccess(data.data));
+    } catch (error) {
+      dispatch(getMembersError(error));
+    }
+  };
+};
+
+export const getMemberById = (id) => {
```
Comment (DelfiLT) [inline src/redux/members/thunks.js:36]: You can't use a getById, you have to search for the element only using the getAll endpoint

**Review by DelfiLT — CHANGES_REQUESTED** [review]

File: src/Components/Members/Create/index.jsx [hunk 7]
```diff
@@ -36,119 +40,52 @@ const MembersCreate = () => {
...
-      phone,
-      email,
-      city,
-      birthDay,
-      postalCode,
-      isActive,
-      membership
-    });
+    dispatch(createMember(member));
+    console.log(member);
```
Comment (DelfiLT) [inline src/Components/Members/Create/index.jsx:77]: Remove console.log

File: src/Components/Members/Create/index.jsx [hunk 8]
```diff
@@ -36,119 +40,52 @@ const MembersCreate = () => {
...
 
   return (
     <div className={styles.container}>
+      <Modal
+        onClose={() => setShowModalError(false)}
+        isOpen={showModalError}
+        title={message}
+        error
+      />
+      ;
```
Comment (DelfiLT) [inline src/Components/Members/Create/index.jsx:88]: Remove this ;

File: src/Components/Members/Create/index.jsx [hunk 9]
```diff
@@ -36,119 +40,52 @@ const MembersCreate = () => {
...
       <Modal onClose={() => setShowModal(false)} isOpen={showModal} title={message} success />;
```
Comment (DelfiLT) [inline src/Components/Members/Create/index.jsx:89]: Remove this ;

File: src/Components/Members/Edit/EditMembers.jsx [hunk 10]
```diff
@@ -99,42 +118,20 @@ const MembersEdit = () => {
...
 
   return (
     <div className={styles.container}>
+      <Modal
+        onClose={() => setShowModalError(false)}
+        isOpen={showModalError}
+        title={message}
+        error
+      />
+      ;
```
Comment (DelfiLT) [inline src/Components/Members/Edit/EditMembers.jsx:134]: Remove this ;

File: src/Components/Members/Edit/EditMembers.jsx [hunk 11]
```diff
@@ -99,42 +118,20 @@ const MembersEdit = () => {
...
       <Modal onClose={() => setShowModal(false)} isOpen={showModal} title={message} success />;
```
Comment (DelfiLT) [inline src/Components/Members/Edit/EditMembers.jsx:135]: Remove this ;

**Review by DelfiLT — APPROVED** [review]
I'll approve this because everything is working and we don't have much time but I'll leave you some UX/UI comments so that you take them into account appart from the code comments.

Try to copy the style from the other componentes, the blue box should have a `border-radius: 1vw` and  `box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);` and it should be sepparated from the nav at the top and delete that blank space. All this should be applied in the forms too.

**Review by ukiweskamp — APPROVED** [review]

File: src/redux/members/reducer.js (hunk 5 above)
Comment (JuanIgnacioSobral) [reply src/redux/members/reducer.js]: Done

File: src/redux/members/thunks.js (hunk 6 above)
Comment (JuanIgnacioSobral) [reply src/redux/members/thunks.js]: Removed

File: src/Components/Members/Create/index.jsx (hunk 7 above)
Comment (JuanIgnacioSobral) [reply src/Components/Members/Create/index.jsx:77]: Removed

File: src/Components/Members/Create/index.jsx (hunk 8 above)
Comment (JuanIgnacioSobral) [reply src/Components/Members/Create/index.jsx:88]: Removed

File: src/Components/Members/Create/index.jsx (hunk 9 above)
Comment (JuanIgnacioSobral) [reply src/Components/Members/Create/index.jsx:89]: Removed

File: src/Components/Members/Edit/EditMembers.jsx (hunk 10 above)
Comment (JuanIgnacioSobral) [reply src/Components/Members/Edit/EditMembers.jsx:134]: Done

File: src/Components/Members/Edit/EditMembers.jsx (hunk 11 above)
Comment (JuanIgnacioSobral) [reply src/Components/Members/Edit/EditMembers.jsx:135]: Also, removed

**Pull request merged**